packages/x-tree-view/src/TreeItem/TreeItem.tsx

```diff
@@ -425,6 +455,7 @@ export const TreeItem = React.forwardRef(function TreeItem(
           {...((enhancedDragAndDropOverlayProps as any).action == null
             ? {}
             : { dragAndDropOverlayProps: enhancedDragAndDropOverlayProps })}
+          labelInputProps={enhancedLabelInputProps}
```

Comment:
If you can do the same thing as for `enhancedDragAndDropOverlayProps`, which is to not pass the props when editing is not enabled. That way we don't have a small BC for people passing `other` to the DOM.

Unless it causes issue here